### DISCUSSION TITLE
transaction model in minio

### DISCUSSION THREAD

**Discussion by lecssmi** [minio/minio#13633] [Q&A] [answered]
s3 achieves eventual consistency, and how about minio?

**Comment by harshavardhana** [comment, marked as answer]
@lecssmi read here https://docs.min.io/docs/distributed-minio-quickstart-guide.html 

> MinIO follows strict read-after-write and list-after-write consistency model for all i/o operations both in distributed and standalone modes. This consistency model is only guaranteed if you use disk filesystems such as xfs, ext4 or zfs etc.. for distributed setup.

**Reply by lecssmi** [reply]
Thanks. Then if I was reading the file, other person performed a delete operation. Will it affect the read operation?

**Reply by harshavardhana** [reply]
> Thanks. Then if I was reading the file, other person performed a delete operation. Will it affect the read operation?

Delete will serialize over Read's() i.e Delete will be honored after Read() is complete.
